### PR TITLE
Add "Block API" initial doc to the documentation app.

### PR DESCRIPTION
**Pull request by mtias**
image

This covers the main properties of block registration and is meant to be a resource highlighting what is available and how it works.

This is not mean to be a tutorial, for which we have a specific section already.

### CODE REVIEW

**Comment by BE-Webdesign** [comment]
This looks great Matias! 👍

**Comment by mtias** [comment]
Thanks! I plan on adding a separate section for `Edit` and `Save` (which would cover `setAttributes`, `focus`, etc), another one for large component utilities (Editable, Toolbar, Inspector, etc), and another one for reusable elements (buttons, form inputs, upload buttons, etc).

Feedback welcome.

File: docs/block-api.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Block API
+
+Blocks are the fundamntal element of the Gutenberg editor. They are the primary way in which plugins and themes can register their own functionality and extend the capabilities of the editor. This document covers the main properties of block registration.
```
Comment (aduth) [inline docs/block-api.md]: Typo: "fundamental"

File: docs/block-api.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#### Category
+
+Blocks are grouped into categories to help users browse and discover them. The core provided categories are `common`, `formatting`, `layout`, `widgets`, and `embeds`.
+
+```js
+// Assigning to the 'layout' category
+category: 'widgets',
+```
+
+#### Icon
```
Comment (aduth) [inline docs/block-api.md]: `icon` is optional (though now that I think of it, I'm not sure how well it appears in the inserter)

File: docs/block-api.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+By default, Gutenberg adds a class with the form `.wp-blocks-your-block-name` to the root element of your saved markup. This helps having a consistent mechanism for styling blocks that themes and plugins can rely on. If for whatever reason a class is not desired on the markup, this functionality can be disabled.
+
+```js
+// Do not generate classes for this block
+className: false,
+```
+
+## Edit and Save
+
+The `edit` and `save` functions define the editor interface with wich a user would interact, and the markup to be serialized back when a post is saved. They are the heart of how a block operates, so we'll cover them separately.
```
Comment (aduth) [inline docs/block-api.md]: Typo: "which"

File: docs/block-api.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Blocks are grouped into categories to help users browse and discover them. The core provided categories are `common`, `formatting`, `layout`, `widgets`, and `embeds`.
+
+```js
+// Assigning to the 'layout' category
+category: 'widgets',
+```
+
+#### Icon
+
+An icon property should be specififed to make it easier to identify a block. These can be any of [WordPress' Dashicons](https://developer.wordpress.org/resource/dashicons/), or a custom `svg` element.
```
Comment (aduth) [inline docs/block-api.md]: Typo: "specified"

File: docs/block-api.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	// Cover image for the book, corresponds to the image src.
+	cover: {
+		type: 'string',
+		source: attr( 'img', 'src' ),
+	},
+	author: {
+		type: 'string',
+		source: children( '.book-author' ),
+	},
+	// This is encoded in the HTML comments of the block and is nto present
```
Comment (aduth) [inline docs/block-api.md]: Typo: "into"

File: docs/block-api.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+Attributes can be quite powerful, if you want to learn more about what's possible, read the specific document.
+
+### Transforms (optional)
+
+Work in progress...
+
+### className (optional)
+
+By default, Gutenberg adds a class with the form `.wp-blocks-your-block-name` to the root element of your saved markup. This helps having a consistent mechanism for styling blocks that themes and plugins can rely on. If for whatever reason a class is not desired on the markup, this functionality can be disabled.
```
Comment (aduth) [inline docs/block-api.md]: Not for much longer if I have my way 😄  #2507

File: docs/block-api.md (hunk 2 above)
Comment (mtias) [reply docs/block-api.md]: We should offer a generic fallback, probably, and keep it optional.

File: docs/block-api.md (hunk 6 above)
Comment (mtias) [reply docs/block-api.md]: ;)

**Review by aduth — APPROVED** [review]

**Pull request merged**